src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java

```diff
@@ -0,0 +1,165 @@
+    bookingRequest.put("submissionDateTime", "2021-11-03T11:41:00+01:00");
+    bookingRequest.put("communicationChannel", "AO");
+    bookingRequest.put("expectedDepartureDate", "2021-05-17");
+    bookingRequest.put("isEquipmentSubstitutionAllowed", true);
+
+    Map<String, Object> commodity = new LinkedHashMap<>();
+    commodity.put("commodityType", "Bloom");
+    commodity.put("cargoGrossWeight", 2000.0);
+    commodity.put("cargoGrossWeightUnit", "LBR");
+    bookingRequest.put("commodities", Arrays.asList(commodity));
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    String jsonBookingRequest = objectMapper.writeValueAsString(bookingRequest);
+
+    return given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .contentType(ContentType.JSON)
+        .body(jsonBookingRequest)
+        .post(Configuration.ROOT_URI + BOOKING_PATH)
+        .then()
+        .assertThat()
+        .statusCode(202)
+        .body(JSON_SCHEMA_VALIDATOR)
+        .extract()
+        .body()
+        .as(Map.class);
+  }
+
+  private List<Map> findAllBookings() {
+    return given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .get(Configuration.ROOT_URI + BOOKING_SUMMARIES_PATH)
+        .then()
+        .assertThat()
+        .statusCode(200)
+        .body("size()", greaterThanOrEqualTo(1))
+        .body(JSON_SCHEMA_VALIDATOR)
+        .extract()
+        .body()
+        .as(List.class);
+  }
+
+  @Test
+  public void testCancelBookingInvalidCarrierBookingRequestReference() {
+    given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .when()
+        .pathParam(
+            "carrierBookingRequestReference",
+            "12345678912345678901235678945651205451686156465154515564845156754845678465544567845648456548151554234")
+        .body(cancellationRequest.toString())
+        .contentType(ContentType.JSON)
+        .patch(Configuration.ROOT_URI + BOOKING_CARRIERBOOKINGREQUESTREFERENCE_PATH)
+        .then()
+        .assertThat()
+        .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void testCancelBookingInvalidRequest() {
+
+    JSONObject cancellation = new JSONObject();
+    cancellation.put("documentStatus", "PENU");
+    cancellation.put("reason", "booking cancelled during test");
+
+    given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .when()
+        .pathParam("carrierBookingRequestReference", "123456789")
+        .body(cancellation.toString())
+        .contentType(ContentType.JSON)
+        .patch(Configuration.ROOT_URI + BOOKING_CARRIERBOOKINGREQUESTREFERENCE_PATH)
+        .then()
+        .assertThat()
+        .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void testCancelBookingInInvalidDocumentStatus() {
+
+    given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .when()
+        .pathParam("carrierBookingRequestReference", bookingsNotAllowedToCancel.get(0))
```

Comment:
Would it not make sense to test for all the given test-data?

Reply:
I changed the test so it executes on all bookings found with an invalid status